daemon/src/connection.rs

```diff
@@ -25,6 +27,8 @@ pub struct Actor {
     /// Max duration since the last heartbeat until we die.
     timeout: Duration,
     connected_state: Option<ConnectedState>,
+    // QUESTION: Does this need to be generic?
```

Comment:
:question: Is it okay to set `wallet::Actor` here?

Reply:
No I think that will become an issue once you try and insert something into this hashmap.

Best to remove the type parameter from `setup_contract` and pass two `MessageChannel`s in. If that works, try and land that as a separate patch :)